### ISSUE TITLE
Can no longer connect to perform an 365 DSC

### ISSUE THREAD

**Issue by martynstar** [microsoft/Microsoft365DSC#2007]
#### Details of the scenario you tried and the problem that is occurring
As im attempting to create a blueprint to verify if tenants are set the same, i updated as we were having issues with assert-blueprint last week,
Now i can not connect at all

#### Verbose logs showing the problem
Connecting to {MicrosoftGraph}...WARNING: Unable to retrieve AccessToken. Have you registered the 'Microsoft Graph PowerShell' application already?
Please run 'Connect-MgGraph -Scopes Domain.Read.All' and logon using username@domain.com

Repeated over and over.

If i type connect-mggraph it pops up asking me to sign in.

The M365DSC commands no longer do this.

#### Suggested solution to the issue
Restore the prompt to enable sign in to 365, this no longer appears.

#### The DSC configuration that is used to reproduce the issue (as detailed as possible)
Assert-M365DSCBlueprint -BluePrintUrl C:\DSC\Blueprint\M365TenantConfig.ps1 -OutputReportPath C:\DSC\0859-13-06-22.html -Credentials $credential
# insert configuration here
```

#### The operating system the target node is running
OsName               : Microsoft Windows Server 2022 Datacenter Azure Edition
OsOperatingSystemSKU : 407
OsArchitecture       : 64-bit
WindowsVersion       : 2009
WindowsBuildLabEx    : 20348.1.amd64fre.fe_release.210507-1500
OsLanguage           : en-US
OsMuiLanguages       : {en-US}
-->

#### Version of the DSC module that was used ('dev' if using current dev branch)
1.22.608.1 Microsoft365DSC
![image](https://user-images.githubusercontent.com/39093867/173313959-a4d370bd-2217-457e-8b43-6157480af962.png)

**Comment by flaviusmp** [comment]
Hello, In my case I have a similar issue. When I run the command Export-M365DSCConfiguration it returns me for the modules "Microsoft Graph" and "Pnp" the following message:
WARNING: Unable to retrieve AccessToken. Have you registered the 'Microsoft Graph PowerShell' application already? Please run 'Connect-MgGraph -Scopes Domain.Read.All'
 and logon using 'user@domain.com'

I am connected already to Microsoft Graph but still keep the error:
PS C:\windows\system32> Connect-MgGraph -Scopes Domain.Read.All
Welcome To Microsoft Graph!

Any suggestion?

**Comment by andikrueger** [comment]
There is a PR on its way that will improve authentication scenarios. see #2005

**Comment by martynstar** [comment]
Awesome! Anytime scales?
I am having to export and import using another tool and comparing as i can no longer use the DSC :(

Thanks

**Comment by OlivierWaterloos** [comment]
When will be the new version be available to resolve this issue?

**Comment by andikrueger** [comment]
It got released yesterday: https://github.com/microsoft/Microsoft365DSC/releases/tag/1.22.615.1

**Comment by martynstar** [comment]
Still the same issue for me
I cant see any logins working at all through any commands
![image](https://user-images.githubusercontent.com/39093867/174116692-0b854c85-b500-47a2-af8f-7580d895aee2.png)

I have removed all the apps, and then allowed it to create at login and still not able to authenticate.

**Comment by desmay** [comment]
@martynstar  So you previously had this all working but one of the latest updates caused issues with connecting to tenant?  Are you using a government clouds (GCC,GCCH,DOD) or commercial cloud ?

**Comment by martynstar** [comment]
Yes!
I've just tried on an older version and it updates settings as you run export.. that works somewhat I can manage to export one setting but it continually prompts to login..
Tomorrow I'll build a fresh PC and run it
Also tried on PSU as well..

If I can roll back a version or two I'd love to try it!

**Comment by martynstar** [comment]
Accidental close

**Comment by desmay** [comment]
And its a commercial cloud ?

**Comment by martynstar** [comment]
Yeah commercial cloud. No government stuff here :)
Was previously working, I had a blueprint ready to test found it missed some data on the export so compiled it manually and tested once.. then an update broke it and then was fixed and now can't extract or blueprint at all

**Comment by andikrueger** [comment]
Please make sure to add the required version to your configuration. 

Import-DscResource Microsoft365dsc -RequiredVersion ….

there was a change and we are searching for that string while parsing the configuration. 

Additionally please verify the Microsoft Graph application permission and registration within your tenant.

**Comment by martynstar** [comment]
Thanks!
I wiped all the powershell modules and started again
This then created new apps in the tenant and still had the same issue!

I will now start from a fresh PC tomorrow to verify it's not made an impact on my live one

I don't actually know what permissions it needs as it always adds automatically, I've updated the permissions using the update commands too.
Also tried connecting using the error string and msgraph connects ok

**Comment by OlivierWaterloos** [comment]
PSDscResources 2.12.0.0  is installed, but I still cannot run Import-DscResource. Which module needs to be installed?

**Comment by martynstar** [comment]
I have just built a fresh windows 11 VM
Installed Microsoft365DSC only nothing else
it installed some pre-reqs
and then reports

![image](https://user-images.githubusercontent.com/39093867/174277108-44d2c3ff-fde5-4aa6-9e99-5d3101b7699e.png)

So I have now deleted the apps in the tenant (hoping they would be automatically created with the usual prompt to do so)
I do not get the Pop up, requesting MFA authentication.
Also No Pop up note on adding the App (list of permissions is usually shown, then tick for admin access etc)

Then ran

Update-M365DSCAllowedGraphScopes -All -Type 'Update' -Environment 'Global'

To attempt to setup the access, this does pop up, creating the app etc (with MFA prompts!)

Then attempting to run the export..
Still the same errors :(

WARNING: Unable to retrieve AccessToken. Have you registered the 'Microsoft Graph PowerShell' application already?

The permission its saying is missing is Domain.read.all - but the app is present
Its the pop up request that causing it, if there is no Pop up for MFA, its never going to authenticate.

Im going to try another tenant now (hundreds to chose from but only one dev)

**Comment by martynstar** [comment]
Same exact message on another tenant..

**Comment by ykuijs** [comment]
Hi all, this message comes from the MSCloudLoginAssistant and is generated by a change I implemented recently. I will test this scenario asap and implement a fix.

@martynstar Could you share the blueprint you are using (remove sensitive info where needed), so I can test this on my environment?

**Comment by martynstar** [comment]
@ykuijs thank you!
I'll get the blueprint edited and share it here, we are only checking for a conditional access policy but any export also fails!

**Comment by mlhickey** [comment]
Seems like the OAuth catch doesn't check for AADSTS50076 (in ErrorDetails, not Exception).  Other helpers catch the STS error and fail into MFA-capable function, e.g.

        if ($_.ErrorDetails -like '*AADSTS50076: Due to a configuration change made by your administrator*')
        {
            Write-Verbose "Connecting to Microsoft Graph with MFA"
            Connect-MSCloudLoginMSGraphWithMFA
            return
        } elseif ($_.Exception -like "System.Net.WebException: The remote server returned an error: (400) Bad Request.*")

**Comment by martynstar** [comment]
Hi!
Any updates on this issue?
Thanks

**Comment by OlivierWaterloos** [comment]
Hello, I saw the new version 1.22.622.1 does not resolve the issue.
Do you have any news?

**Comment by ykuijs** [comment]
Hi guys, I have been Out-of-the-Office last week. Am working on this issue as we speak. Hopefully will be able to find the root cause quickly and implement a fix asap.

**Comment by ykuijs** [comment]
Have been able to reproduce the warning message, now working on finding the root cause!

**Comment by ykuijs** [comment]
Found root cause, will fix the issue and submit a PR!

**Comment by ykuijs** [comment]
Closed via #2039, will be included in the next release

**Comment by OlivierWaterloos** [comment]
The version 1.22.629.1 is working fine to connect to MS-Graph 
Thanks
